api/lib/domain/models/CertificationAssessment.js

```diff
@@ -111,6 +115,22 @@ class CertificationAssessment {
   getChallengeRecIdByQuestionNumber(questionNumber) {
     return this.certificationAnswersByDate[questionNumber - 1]?.challengeId || null;
   }
+
+  skipUnansweredChallenges() {
+    this.certificationChallenges.map((certificationChallenge) => {
+      if (!this.certificationAnswersByDate.some((certificationAnswer) => certificationChallenge.challengeId === certificationAnswer.challengeId)) {
+        return certificationChallenge.skipAutomatically();
+      }
+    });
+  }
+
+  neutralizeUnansweredChallenges() {
+    this.certificationChallenges.map((certificationChallenge) => {
+      if (!this.certificationAnswersByDate.some((certificationAnswer) => certificationChallenge.challengeId === certificationAnswer.challengeId)) {
+        return certificationChallenge.neutralize();
+      }
+    });
+  }
 }
```

Comment:
Lorsqu’on utilise pas la valeur de retour d’un map, ca veut à priori dire qu’on ne devrait pas utiliser un map!
Le map devrait être une fonction pure. Ici on a un effet de bord `skip`

```suggestion
  skipUnansweredChallenges() {
    this.certificationChallenges.forEach((certificationChallenge) => {
      if (!this.certificationAnswersByDate.some((certificationAnswer) => certificationChallenge.challengeId === certificationAnswer.challengeId)) {
       certificationChallenge.skipAutomatically();
      }
    });
  }

  neutralizeUnansweredChallenges() {
    this.certificationChallenges.forEach((certificationChallenge) => {
      if (!this.certificationAnswersByDate.some((certificationAnswer) => certificationChallenge.challengeId === certificationAnswer.challengeId)) {
        certificationChallenge.neutralize();
      }
    });
  }
}
```